flocker/testtools/test/test_flaky.py

```diff
@@ -322,3 +297,68 @@ def test_delayed(self):
         result = ExtendedTestResult(log)
         test.run(result)
         self.assertThat(test.log, Equals([('startTest', test)]))
+
+
+class FlakyIntegrationTests(testtools.TestCase):
+    """
+    Tests that @flaky and retry_flaky interact well with our standard base
+    test cases.
+    """
+
+    @given(base_test_cases, jira_keys, num_runs, num_runs,
+           streaming(text(average_size=10)).map(iter))
+    def test_flaky_skipped_test(self, base_test_case, jira_keys, max_runs,
+                                min_passes, reasons):
+        """
+        If a test is skipped and also marked @flaky, we report it as skipped.
+
+        Ensures that our workarounds for testing-cabal/unittest-ext#60
+        work with @flaky support.
+        """
+        [min_passes, max_runs] = sorted([min_passes, max_runs])
+        observed_reasons = []
+
+        class SkippingTest(base_test_case):
+            run_tests_with = retry_flaky()
+
+            @flaky(jira_keys, max_runs, min_passes)
+            def test_skip(self):
+                observed_reasons.append(reasons.next())
+                raise unittest.SkipTest(observed_reasons[-1])
+
+        test = SkippingTest('test_skip')
+        self.assertThat(run_test(test), only_skips(1, observed_reasons))
+
+    @given(base_test_cases, jira_keys, num_runs, num_runs)
+    def test_retry_by_default(self, base_test_case, jira_keys, max_runs,
+                              min_passes):
+        """
+        Tests that inherit from our base test cases and use @flaky are retried
+        by default.
+        """
+        [min_passes, max_runs] = sorted([min_passes, max_runs])
+
+        class ErroringTest(base_test_case):
+            @flaky(jira_keys, max_runs, min_passes)
+            def test_error(self):
+                1/0
+
+        test = ErroringTest('test_error')
+        self.assertThat(
+            run_test(test),
+            has_results(
+                tests_run=Equals(1),
+                errors=MatchesListwise([
```

Comment:
I'm not sure that I understand what this is supposed to be matching, or how this is related to the test description. Please add a comment explaining what a sample expected results looks like, and why that indicates that the test has been retried.